### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by lishaduck**
So that I can run `turbo watch build`. Bumped the rest while I was at it.

### CODE REVIEW

**Comment by lishaduck** [comment]
I bumped everything that wouldn't make external consumers need to bump.
I guess if #157 #156 are majors, then it doesn't matter, but I left that up to you.

**Comment by lishaduck** [comment]
Oh, and I also noticed some publint complaints, so I bumped tsup and fixed your bundle.

File: .vscode/settings.json [hunk 1]
```diff
@@ -21,5 +21,7 @@
       "url": "https://cdn.jsdelivr.net/npm/tsup/schema.json",
       "fileMatch": ["package.json", "tsup.config.json"]
     }
-  ]
+  ],
+  "typescript.tsdk": "node_modules/typescript/lib",
```
Comment (AndreaPontrandolfo) [inline .vscode/settings.json:25]: we don't have Typescript at root, so i think this wouldn't take effect.

File: turbo.json [hunk 2]
```diff
@@ -1,33 +1,61 @@
 {
-  "$schema": "https://turbo.build/schema.v1.json",
-  "globalDependencies": ["**/.env.*local"],
-  "pipeline": {
+  "$schema": "https://turbo.build/schema.json",
+  "globalDependencies": [
```
Comment (AndreaPontrandolfo) [inline turbo.json:3]: why this?

File: packages/sheriff-types/package.json [hunk 3]
```diff
@@ -7,14 +7,12 @@
...
   ],
-  "main": "./src/index.ts",
-  "types": "./src/index.ts",
   "module": "./src/index.ts",
+  "types": "./src/index.ts",
   "exports": {
     ".": {
       "types": "./src/index.ts",
-      "import": "./src/index.ts",
-      "require": "./src/index.ts"
```
Comment (AndreaPontrandolfo) [inline packages/sheriff-types/package.json:17]: why are we removing require? Wouldn't this break CJS builds?

File: packages/eslint-config-sheriff/package.json [hunk 4]
```diff
@@ -2,15 +2,21 @@
   "name": "eslint-config-sheriff",
   "version": "18.7.0",
   "description": "A comprehensive and opinionated Typescript-first ESLint configuration.",
-  "main": "./dist/index.js",
+  "main": "./dist/index.cjs",
```
Comment (AndreaPontrandolfo) [inline packages/eslint-config-sheriff/package.json:5]: i thunk we can probably just removing this

File: package.json [hunk 5]
```diff
@@ -16,7 +16,8 @@
     "typesync": "turbo run typesync",
     "knip": "knip",
     "validate-config": "turbo run validate-config",
-    "merge-checks": "turbo run publint manypkg typesync knip typecheck lint validate-config",
+    "merge-checks": "turbo run publint manypkg typesync knip typecheck lint validate-config check-deduped-deps",
+    "check-deduped-deps": "pnpm dedupe --check",
```
Comment (AndreaPontrandolfo) [inline package.json:20]: what's the point of deduping?

File: .prettierignore [hunk 6]
```diff
@@ -31,6 +31,7 @@ yarn-error.log*
 
 # turbo
 .turbo
+turbo.json
```
[the remark this answers is not in the document]
Comment (lishaduck) [reply .prettierignore:34]: @turbo/codemod doesn't keep formatting. I usually gitignore it, but you could just reformat it if you prefer.

File: .vscode/settings.json [hunk 7]
```diff
@@ -21,5 +21,7 @@
...
+  "eslint.useFlatConfig": true
```
[the remark this answers is not in the document]
Comment (lishaduck) [reply .vscode/settings.json:26]: Not for eslint 8.

File: .vscode/settings.json (hunk 1 above)
Comment (lishaduck) [reply .vscode/settings.json:25]: 🤦‍♂️ Yup.

File: turbo.json (hunk 2 above)
Comment (lishaduck) [reply turbo.json:3]: @turbo/codemod reformatted it 🤷‍♂️

File: packages/sheriff-types/package.json (hunk 3 above)
Comment (lishaduck) [reply packages/sheriff-types/package.json:17]: Oh, is sheriff-types published? Oh, yes it is. As it wasn't bundled, I assumed it was internal only.

File: packages/eslint-config-sheriff/package.json (hunk 4 above)
Comment (lishaduck) [reply packages/eslint-config-sheriff/package.json:5]: Yeah. I think exports overrides it, but I didn't want to remove it if it was there intentionally.

File: package.json (hunk 5 above)
Comment (lishaduck) [reply package.json:20]: If you've got `foo@1.0` which depends on `baz@1.0`, and then `baz` gets a new release, then install `bar@1.0` which depends on `baz@1.1`, then you'll get two `baz`es, as it won't upgrade the other baz. Then, PNPM will end up downloading and install many different copies over time, which is both wasteful and can break types.

File: .prettierignore (hunk 6 above)
[the remark this answers is not in the document]
Comment (AndreaPontrandolfo) [reply .prettierignore:34]: Ya, we want to keep formatting with Prettier, let's remove this change

File: .vscode/settings.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (AndreaPontrandolfo) [reply .vscode/settings.json:26]: Oh ok. Oh well, it doesn't hurt either way, so it's fine.

File: packages/sheriff-types/package.json (hunk 3 above)
Comment (AndreaPontrandolfo) [reply packages/sheriff-types/package.json:17]: Oh wait, you mean that there is no runtime code, so we can just export the `"types": "./src/index.ts"` and don't need the rest. I think you are right, I just copy/pasted that part of the config from the other packages. If this work as before, i'm fine with this change.

File: packages/eslint-config-sheriff/package.json (hunk 4 above)
Comment (AndreaPontrandolfo) [reply packages/eslint-config-sheriff/package.json:5]: Ya so, `exports` overrides it, but i think it is used for some backward compatibility.
I don't really mind either way, i thought `"main": "./dist/index.js"` was fine, not sure what `publint` says in this case.

File: packages/sheriff-types/package.json (hunk 3 above)
Comment (lishaduck) [reply packages/sheriff-types/package.json:17]: The CJS/ESM distinction still matters with types.
However, the types won't be required or imported, they're only types.

File: turbo.json [hunk 8]
```diff
@@ -36,6 +36,17 @@
...
     },
-    "serve": { "cache": false, "persistent": true }
+    "serve": {
+      "cache": false,
+      "persistent": true
+    },
+    "check-deduped-deps": {
+      "inputs": [
+        "package.json",
+        "apps/*/package.json",
```
Comment (AndreaPontrandolfo) [inline turbo.json:46]: shouldn't this be 
`apps/*/package.json` --> `apps/**/package.json`
`packages/*/package.json` --> `packages/**/package.json`

**Comment by AndreaPontrandolfo** [comment]
Also we are missing the changeset here

File: turbo.json (hunk 8 above)
Comment (lishaduck) [reply turbo.json:46]: Do you have nested `package.json`s? Maybe?

Comment (AndreaPontrandolfo) [reply turbo.json:46]: mmm, no, we just have 1 package.json per workspace.

**Comment by lishaduck** [comment]
Should `typesync` v0.13.0 be included? It's an easy migration (nothing breaking, hopefully. If so, that's a bug.). I added native PNPM monorepo support, so it'd require tweaking tasks though and this is big. And I don't really want to clone Sheriff on this machine.

**Comment by AndreaPontrandolfo** [comment]
I saw your PR, it's great, we use typesync at my workplace. I'm gonna try the update!
For Sheriff, yeah, I'm gonna add it to the backlog, we'll see.

**Comment by lishaduck** [comment]
> I saw your PR, it's great, we use typesync at my workplace. I'm gonna try the update! For Sheriff, yeah, I'm gonna add it to the backlog, we'll see.

Ok, I don't really care when it gets bumped, but if you do bump, I'd suggest basing it on this branch.

**Review by AndreaPontrandolfo — APPROVED** [review]

**Pull request merged**